homeassistant/components/squeezebox_player/media_player.py

```diff
@@ -0,0 +1,218 @@
+        # we delay adding the player a small bit because the player name may be received
+        # just a bit after connect. This way we can create a device reg entry with the correct name
+        for _ in range(10):
+            if player.player_id not in player.name:
+                break
+            await asyncio.sleep(0.1)
+        added_ids.add(player.player_id)
+        async_add_entities([SqueezeboxPlayer(slimserver, player)])
+
+    async def on_slim_event(event: SlimEvent):
+        """Handle player added/connected event."""
+        player = slimserver.get_player(event.player_id)
+        await async_add_player(player)
+
+    # register listener for new players
+    config_entry.async_on_unload(
+        slimserver.subscribe(on_slim_event, EventType.PLAYER_CONNECTED)
+    )
+
+    # add all current items in controller
+    for player in slimserver.players:
+        await async_add_player(player)
+
+
+class SqueezeboxPlayer(MediaPlayerEntity):
+    """Representation of MediaPlayerEntity from Squeezebox Player."""
+
+    def __init__(self, slimserver: SlimServer, player: SlimClient) -> None:
+        """Initialize MediaPlayer entity."""
+        self.slimserver = slimserver
+        self.player = player
+        self._attr_should_poll = False
+        self._attr_supported_features = SUPPORTED_FEATURES
+        self._attr_device_class = MediaPlayerDeviceClass.SPEAKER
+
+        self.update_attributes()
+
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+        self.update_attributes()
+        self.async_on_remove(
+            self.slimserver.subscribe(
+                self._on_slim_event,
+                (
+                    EventType.PLAYER_UPDATED,
+                    EventType.PLAYER_CONNECTED,
+                    EventType.PLAYER_DISCONNECTED,
+                    EventType.PLAYER_NAME_RECEIVED,
+                    EventType.PLAYER_RPC_EVENT,
+                ),
+                player_filter=self.player.player_id,
+            )
+        )
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return unique id for entity."""
+        return self.player.player_id
+
+    @property
+    def available(self) -> bool:
+        """Return availability of entity."""
+        return self.player.connected
+
+    @property
+    def state(self) -> str:
+        """Return current state."""
+        if not self.player.powered:
+            return STATE_OFF
+        return STATE_MAPPING[self.player.state]
+
+    @callback
+    def update_attributes(self) -> None:
+        """Handle player updates."""
+        self._attr_name = self.player.name
+        self._attr_volume_level = self.player.volume_level / 100
+        self._attr_media_position = self.player.elapsed_seconds
+        self._attr_media_position_updated_at = utcnow()
+        self._attr_media_content_id = self.player.current_url
+        self._attr_media_content_type = "music"
+        self._attr_device_info = DeviceInfo(
```

Comment:
This is only fetched during startup so we don't have to generate it all the time. Let's just set this in the constructor.

Reply:
Yes, also leftover from me experimenting with the name arriving later but finally settled on the initial wait